### PR TITLE
Polish

### PR DESCRIPTION
**Pull request by izeye**
This PR fixes some typos and polishes trivial stuff.

### CODE REVIEW

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 /**
  * {@link org.springframework.cache.Cache} implementation on top of a
- * {@link javax.cache.Cache} instance.
```
Comment (sbrannen) [inline spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: Without the fully qualified package name, we'd want to have "JCache" there in addition.

So we might as well just leave it as is for clarity, as was originally intended.

Thus, please either undo this change or prefix that with "JCache" as is done elsewhere in the Javadoc for this class.

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java [hunk 2]
```diff
@@ -42,15 +42,15 @@
 
 
 	/**
-	 * Create an {@link org.springframework.cache.jcache.JCacheCache} instance.
+	 * Create a {@link org.springframework.cache.jcache.JCacheCache} instance.
```
Comment (sbrannen) [inline spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: The package name starts with an "o". Thus this is grammatically correct.

Please undo this change.

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java [hunk 3]
```diff
@@ -28,7 +28,7 @@
 
 /**
  * {@link org.springframework.cache.Cache} implementation on top of a
- * {@link javax.cache.Cache} instance.
+ * {@link Cache} instance.
```
Comment (snicoll) [inline spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: That full package name is actually on purpose as the Spring Cache abstraction has also a concept of `Cache`.

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java [hunk 4]
```diff
@@ -42,15 +42,15 @@
...
+	 * Create a {@link org.springframework.cache.jcache.JCacheCache} instance.
 	 * @param jcache backing JCache Cache instance
 	 */
 	public JCacheCache(Cache<Object, Object> jcache) {
 		this(jcache, true);
 	}
 
 	/**
-	 * Create an {@link org.springframework.cache.jcache.JCacheCache} instance.
+	 * Create a {@link org.springframework.cache.jcache.JCacheCache} instance.
```
Comment (sbrannen) [inline spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: The package name starts with an "o". Thus this is grammatically correct.

Please undo this change.

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCacheManager.java [hunk 5]
```diff
@@ -28,7 +28,7 @@
 
 /**
  * {@link org.springframework.cache.CacheManager} implementation
- * backed by a JCache {@link javax.cache.CacheManager}.
+ * backed by a JCache {@link CacheManager}.
```
Comment (snicoll) [inline spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCacheManager.java]: Ditto

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCacheManager.java [hunk 6]
```diff
@@ -53,22 +53,22 @@ public JCacheCacheManager() {
 
 	/**
 	 * Create a new JCacheCacheManager for the given backing JCache.
-	 * @param cacheManager the backing JCache {@link javax.cache.CacheManager}
+	 * @param cacheManager the backing JCache {@link CacheManager}
```
Comment (snicoll) [inline spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCacheManager.java]: And here

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCacheManager.java [hunk 7]
```diff
@@ -53,22 +53,22 @@ public JCacheCacheManager() {
...
 	 */
 	public JCacheCacheManager(CacheManager cacheManager) {
 		this.cacheManager = cacheManager;
 	}
 
 
 	/**
-	 * Set the backing JCache {@link javax.cache.CacheManager}.
+	 * Set the backing JCache {@link CacheManager}.
```
Comment (snicoll) [inline spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCacheManager.java]: etc etc

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCacheManager.java [hunk 8]
```diff
@@ -53,22 +53,22 @@ public JCacheCacheManager() {
...
 	 */
 	public void setCacheManager(@Nullable CacheManager cacheManager) {
 		this.cacheManager = cacheManager;
 	}
 
 	/**
-	 * Return the backing JCache {@link javax.cache.CacheManager}.
+	 * Return the backing JCache {@link CacheManager}.
```
Comment (snicoll) [inline spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCacheManager.java]: That too

**Review by snicoll — COMMENTED** [review]
Thanks for the PR @izeye! I've added a few suggestions to better explain the rationale. Does that make sense?

**Review by sbrannen — DISMISSED** [review]

**Comment by sbrannen** [comment]
@snicoll, it looks like our reviews crossed paths in the either. Oops! 😉

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java (hunk 3 above)
Comment (izeye) [reply spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: @snicoll If so, it should be `{@link Cache javax.cache.Cache}`, not `{@link javax.cache.Cache}` as the latter will be rendered as just `Cache` in Javadoc. Is it okay to change to `{@link Cache javax.cache.Cache}`?

See https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/cache/jcache/JCacheCache.html

**Comment by sbrannen** [comment]
@snicoll, after our combined reviews, it appears that the only remaining polishing would be the removal of the extra space in the exception message.

Thus, I've dismissed my review and will let you take it from here.

Cheers,

Sam

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java (hunk 1 above)
Comment (izeye) [reply spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: @sbrannen It looks the same concern as @snicoll 's https://github.com/spring-projects/spring-framework/pull/1938#discussion_r212008591. I answered there and let me know what you think.

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java (hunk 2 above)
Comment (izeye) [reply spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: @sbrannen "a" is intended for the rendered link in Javadoc, "JCacheCache", not "org.springframework.cache.jcache.JCacheCache". Am I missing something?

See https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/cache/jcache/JCacheCache.html

Comment (sbrannen) [reply spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: Hmmm.... good question!

I believe we (i.e., the collective core Spring Team) were thinking that the fully qualified class name (FQCN) shows up in the generated HTML when linked via FQCN.

So, if that's not the case, that would mean there is a larger issue across the entire code base in the core Spring Framework, since we routinely use "an" instead of "a" in such cases.

Perhaps it's better to address that nuanced feature of Javadoc in a dedicated JIRA issue.

@snicoll, thoughts?

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java (hunk 3 above)
Comment (snicoll) [reply spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: Oh dear, good catch. Yes please.

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java (hunk 2 above)
Comment (snicoll) [reply spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: @sbrannen thanks. Could you please raise a SPR issue for this? I got caught as well and I am wondering if that's not a new feature of javadoc or something (or I've been wrong this all time).

**Comment by izeye** [comment]
@snicoll @sbrannen Thanks for the feedback! I've updated this PR as suggested.

**Review by snicoll — APPROVED** [review]

**Comment by snicoll** [comment]
Thanks a lot @izeye, this is now merged in `master`.

File: spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java (hunk 2 above)
Comment (sbrannen) [reply spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: Indeed! I am also wondering if this was a change to how Javadoc works in a recent release (perhaps in JDK 8?).

And well... if you've been wrong this whole time, so have I and many others. 😉 

I'll raise a JIRA issue to track this.

Comment (sbrannen) [reply spring-context-support/src/main/java/org/springframework/cache/jcache/JCacheCache.java]: Done: https://jira.spring.io/browse/SPR-17208
